packages/nextjs/hooks/scaffold-eth/useScaffoldContractRead.ts

```diff
@@ -0,0 +1,24 @@
+import { useContractRead, useNetwork } from "wagmi";
+import type { Abi } from "abitype";
+import { getDeployedContract } from "~~/components/scaffold-eth/Contract/utilsContract";
+
+/**
+ * @param contractName - deployed contract name
+ * @param functionName - name of the function to be called
+ * @param readConfig   - wagmi configurations
+ */
+export const useScaffoldContractRead = (
+  contractName: string,
+  functionName: string,
+  readConfig?: Parameters<typeof useContractRead>[0],
+) => {
+  const { chain } = useNetwork();
+  const deployedContractData = getDeployedContract(chain?.id.toString(), contractName);
+
+  return useContractRead({
+    ...readConfig,
+    functionName,
+    address: deployedContractData?.address,
+    abi: deployedContractData?.abi as Abi,
```

Comment:
Can we make some watch / polling magic here?

Some ideas:
- `watch: true` when on hardhat
- Polling time in live networks.

Some inspiration from `useEthPrice`:
```js
const pollingTime = process.env.NEXT_PUBLIC_RPC_POLLING_INTERVAL
  ? parseInt(process.env.NEXT_PUBLIC_RPC_POLLING_INTERVAL)
  : 30_000;
```

Reply:
Just added `wath:true` at [0a446cf](https://github.com/scaffold-eth/se-2/pull/93/commits/0a446cff3746939be2b995a4e560bf22aa1a6649) because we have already configured  pollingInterval for `watch:true`  at https://github.com/scaffold-eth/se-2/pull/45 🙌

Reply:
True that!!! I forgot. Thanks :)